src/svg.rs

```diff
@@ -33,13 +35,37 @@ impl SvgVersion {
 declare_surface!(SvgSurface, SurfaceType::Svg);
 
 impl SvgSurface {
-    pub fn new<P: AsRef<Path>>(width: f64, height: f64, path: P) -> Result<SvgSurface, Status> {
-        let path = path.as_ref().to_string_lossy().into_owned();
-        let path = CString::new(path).unwrap();
+    pub fn new<P: AsRef<Path>>(
+        width: f64,
+        height: f64,
+        path: Option<P>,
+    ) -> Result<SvgSurface, Status> {
+        #[cfg(not(windows))]
+        let path = path.map(|p| {
+            CString::new(p.as_ref().as_os_str().as_bytes()).expect("Invalid path with NULL bytes")
+        });
+        #[cfg(windows)]
+        let path = path.map(|p| {
+            let path_str = p
+                .as_ref()
+                .to_str()
+                .expect("Path can't be represented as UTF-8")
+                .to_owned();
+            if path_str.starts_with("\\\\?\\") {
+                CString::new(path_str[4..].as_bytes())
+            } else {
+                CString::new(path_str.as_bytes())
+            }
+            .expect("Invalid path with NUL bytes")
+        });
 
         unsafe {
             Ok(Self(Surface::from_raw_full(
-                ffi::cairo_svg_surface_create(path.as_ptr(), width, height),
+                ffi::cairo_svg_surface_create(
+                    path.map(|p| p.as_ptr()).unwrap_or(ptr::null()),
```

Comment:
Windows works with `path.as_ref().map(|p| p.as_ptr()).unwrap_or(ptr::null()),`

Reply:
Ok yeah, not doing `as_ref()` will consume the `CString` here and we have a dangling pointer again. Not just for Windows.